### DISCUSSION TITLE
What is the recommended way to get the unminified code?

### DISCUSSION THREAD

**Discussion by csustan** [AR-js-org/AR.js#430] [Help] [answered]
What is the recommended way to get the unminified code for the entire project?

**Comment by kalwalt** [comment, marked as answer]
@csustan the project use webpack as a bundler, you can rebuild the libs by yourself with the `dev` option, run this command `npm run build:dev`. Anyway this is intended only for testing not for production library builds.

**Reply by csustan** [reply]
When I try `npm run build:dev` from the root directory of a freshly cloned repo, I get the message:
`sh: ./node_modules/.bin/webpack: No such file or directory
`
If I attempt it from an instance that was installed with the `npm i @ar-js-org/ar.js `command, I get the error
`npm ERR! Missing script: "build:dev"`

**Reply by kalwalt** [reply]
```
When I try npm run build:dev from the root directory of a freshly cloned repo, I get the message:
sh: ./node_modules/.bin/webpack: No such file or directory
```
This may happens if you have not installed all the dependencies, before to run the build:dev script you need to install all the dependencies with `npm install`.

```
If I attempt it from an instance that was installed with the npm i @ar-js-org/ar.js command, I get the error
npm ERR! Missing script: "build:dev"
```
I don't know if this is allowed or not, but i recommend you to clone the AR.js repository, install the dependencies and build the project with build:dev command.

**Reply by csustan** [reply]
I will try these steps and report back.

**Reply by csustan** [reply]
This must be a problem with my Mac -- `NPM Install` didn't solve the issue, but cloning the repo, then running `npm install webpack-dev-server` followed by `npm run build:dev` seems to have done the trick.
Thank you for your help, @kalwalt
